lib/trace/core.js

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var events = require('events');
+var util = require('util');
+
+var _ = require('underscore');
+
+
+var eventConfigMap = {
+    'membership.update': {
+        sourcePath: ['membership', 'updated'],
+        sinkOptsDefaults: {
+            log: {
+                prefix: '/trace/membership.update'
+            },
+            tchannel: {
+                endpoint: '/trace/membership.update'
+            }
+        }
+    }
+};
+
+// resolves and returns a tracer config object, with
+// extEvent, sourceEmitter, sourceEvent
+function resolveTracerConfig(ringpop, extEvent) {
+    var eventConfig = eventConfigMap[extEvent];
+    var emitterPath = eventConfig.sourcePath;
```

Comment:
Absolutely true, and good grasp :) I wrote this as an assert/return combo to check myself, but didn't clean it properly. Gone.